kratos/containers/dof_variables_container.h

```diff
@@ -0,0 +1,292 @@
+    }
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    virtual std::string Info() const
+    {
+        std::stringstream buffer;
+        buffer << "DofVariablesContainer";
+        return buffer.str();
+    }
+
+    /// Print information about this object.
+    virtual void PrintInfo(std::ostream& rOStream) const
+    {
+        rOStream << "DofVariablesContainer";
+    }
+
+    /// Print object's data.
+    virtual void PrintData(std::ostream& rOStream) const
+    {
+        rOStream << "Double Variables:" << std::endl;
+        for (DoubleVariablesContainerType::const_iterator it = mDofDoubleVariables.begin();
+            it != mDofDoubleVariables.end(); ++it)
+        {
+            (*it)->PrintInfo(rOStream);
+            rOStream << std::endl;
+        }
+        rOStream << "Components Variables:" << std::endl;
+        for (ComponentsVariableContainerType::const_iterator it = mDofComponentsVariables.begin();
+            it != mDofComponentsVariables.end(); ++it)
+        {
+            (*it)->PrintInfo(rOStream);
+            rOStream << std::endl;
+        }
+    }
+
+    ///@}
+
+private:
+    ///@name Member Variables
+    ///@{
+
+    /// Container for double variables
+    DoubleVariablesContainerType mDofDoubleVariables;
+
+    /// Container for variable components
+    ComponentsVariableContainerType mDofComponentsVariables;
+
+    ///@}
+    ///@name Serialization
+    ///@{
+
+    friend class Serializer;
+
+    virtual void save(Serializer& rSerializer) const
+    {
+        std::size_t DoubleVariablesSize= mDofDoubleVariables.size();
+        rSerializer.save("DoubleVariablesSize",DoubleVariablesSize);
+        for(std::size_t i = 0; i < DoubleVariablesSize; i++)
+            rSerializer.save("Variable Name", mDofDoubleVariables[i]->Name());
+
+        std::size_t ComponentsVariablesSize= mDofComponentsVariables.size();
+        rSerializer.save("ComponentsVariablesSize", ComponentsVariablesSize);
+        for(std::size_t i = 0; i < ComponentsVariablesSize; i++)
+            rSerializer.save("Variable Name", mDofComponentsVariables[i]->Name());
+    }
+
+    virtual void load(Serializer& rSerializer)
+    {
+        std::string Name;
+        std::size_t DoubleVariablesSize;
+        rSerializer.load("DoubleVariablesSize",DoubleVariablesSize);
+        for(std::size_t i = 0; i < DoubleVariablesSize; i++)
+        {
+            rSerializer.load("Variable Name", Name);
+            Add( *(static_cast<DoubleVariableType*>(KratosComponents<VariableData>::pGet(Name))) );
+        }
+
+        std::size_t ComponentsVariablesSize;
+        rSerializer.load("ComponentsVariablesSize", ComponentsVariablesSize);
+        for(std::size_t i = 0; i < ComponentsVariablesSize; i++)
+        {
+            rSerializer.load("Variable Name", Name);
+            Add( *(static_cast<VariableComponentType*>(KratosComponents<VariableData>::pGet(Name))) );
```

Comment:
I don't understand this cast, can you explain please what are you doing?. In theory references and pointers are converted with dynamic_cast